flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -374,7 +378,7 @@ void createIcebergTable(ObjectPath tablePath, CatalogBaseTable table, boolean ig
       throws CatalogException, TableAlreadyExistException {
     validateFlinkTable(table);
 
-    Schema icebergSchema = FlinkSchemaUtil.convert(table.getSchema());
+    Schema icebergSchema = FlinkSchemaUtil.convert(((ResolvedCatalogTable) table).getResolvedSchema());
```

Comment:
is it always safe to type cast to `ResolvedCatalogTable`?

Reply:
yes, in this interface, we always get `ResolvedCatalogTable`